addons/hr_timesheet/views/hr_timesheet_portal_templates.xml

```diff
@@ -33,14 +33,24 @@
             </t>
             <t t-if="grouped_timesheets">
                 <t t-call="portal.portal_table">
+                    <thead>
+                        <tr>
+                            <th t-if="not groupby == 'date'">Date</th>
+                            <th t-if="not groupby == 'employee'">Employee</th>
+                            <th t-if="not groupby == 'project'">Project</th>
+                            <th t-if="not groupby == 'task'">Task</th>
+                            <th>Description</th>
+                            <th t-if="is_uom_day" class="text-right">Days Spent</th>
+                            <th t-else="" class="text-right">Hours Spent</th>
+                        </tr>
+                    </thead>
                     <t t-foreach="grouped_timesheets" t-as="timesheets_with_hours">
                         <t t-set="timesheets" t-value="timesheets_with_hours[0]"/>
                         <t t-set="hours_spent" t-value="timesheets_with_hours[1]"/>
                         <thead style="font-size: 0.8rem">
```

Comment:
I think there is something wrong here, isn't it?
This will generate something like this:
```html
<thead>
<!-- len(grouped_timesheets) times what's between here -->
<thead>
<tbody>
<!-- and here ... -->
```
So you would get two `thead` for the first `tbody` which is not valid.
This article may help you in order to solve this https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody#multiple_bodies